### DISCUSSION TITLE
CUE bytes types and the !!binary yaml tag

### DISCUSSION THREAD

**Discussion by PierreR** [cue-lang/cue#1465] [Q&A]
Since v0.4.1 the yaml binary tag `!!binary` is added in generated k8s secret  yaml resources. 

In general it is added for all `bytes` types. So the following input 

```
password: "hello"
base64encoded_password: 'aGVsbG8='
```
produces:

```
password: hello
base64encoded_password: !!binary YUdWc2JHOD0=
```

That feels odd because `YUdWc2JHOD0=` is "hello" twice base64 encoded. At the end, there is no harm because it will be decoded twice.

https://tip.cuelang.org/play/#cue@export@yaml

I would love to understand a bit more why this double encoding is the right thing to do. 

Cheers,

**Comment by verdverm** [comment]
There were previous inconsistencies between how json & yaml handled binary. JSON base64 encoded while yaml did not. They look to now both be doing the base64 encoding on the value in CUE -> <format>

You can do the following now

```
password: "hello"
base64encoded_password: 'hello'
```

https://tip.cuelang.org/play/?id=zVsGkT234va#cue@export@yaml (tip, probably close to v0.4.1) (shows consistency)

https://cuelang.org/play/?id=zVsGkT234va#cue@export@json (v0.4.0 CUE playground) (shows difference) (accessed on Jan 16, 2022)
